mrp_production_batch/models/mrp_production_batch.py

```diff
@@ -0,0 +1,226 @@
+
+    total_production_count = fields.Integer(
+        compute="_compute_total_production_count",
+    )
+
+    def _compute_total_production_count(self):
+        for rec in self:
+            rec.total_production_count = len(rec.production_ids)
+
+    to_review_production_count = fields.Integer(
+        compute="_compute_production_wo_batch_count",
+    )
+    ready_production_count = fields.Integer(
+        compute="_compute_production_wo_batch_count",
+    )
+
+    def _compute_production_wo_batch_count(self):
+        for rec in self:
+            rec.ready_production_count = len(
+                rec.production_ids.filtered(lambda r: r.is_ready_to_produce)
+            )
+            rec.to_review_production_count = len(
+                rec.production_ids.filtered(lambda r: not r.is_ready_to_produce)
+            )
+
+    creation_date = fields.Date(
+        string="Creation Date",
+    )
+    state = fields.Selection(
+        selection=[
+            ("draft", "Draft"),
+            ("in_progress", "In Progress"),
+            ("done", "Done"),
+        ],
+        compute="_compute_state",
+        store=True,
+    )
+
+    @api.depends("production_ids.is_ready_to_produce", "production_ids.state")
+    def _compute_state(self):
+        for rec in self:
+            if rec.ready_production_count > 0:
+                if rec.production_ids.filtered(lambda r: r.state not in ["done"]):
+                    rec.state = "in_progress"
+                else:
+                    rec.state = "done"
+            else:
+                rec.state = "draft"
+
+    operation_type = fields.Many2one(
+        comodel_name="stock.picking.type",
+        required=True,
+    )
+
+    product_ids = fields.Many2many(
+        comodel_name="product.product",
+        compute="_compute_product_ids",
+        string="Products",
+    )
+
+    def _compute_product_ids(self):
+        for rec in self:
+            rec.product_ids = rec.production_ids.product_id
+
+    ready_production_ids = fields.Many2many(
+        comodel_name="mrp.production",
+        compute="_compute_production_ready_ids",
+    )
+
+    def _compute_production_ready_ids(self):
+        for rec in self:
+            rec.ready_production_ids = rec.production_ids.filtered(
+                lambda x: x.is_ready_to_produce
+            )
+
+    to_review_production_ids = fields.Many2many(
+        comodel_name="mrp.production",
+        compute="_compute_production_to_review_ids",
+    )
+
+    def _compute_production_to_review_ids(self):
+        for rec in self:
+            rec.to_review_production_ids = rec.production_ids.filtered(
+                lambda x: not x.is_ready_to_produce
+            )
+
+    def action_check(self):
+        self.ensure_one()
+        productions = self.production_ids.filtered(lambda r: not r.is_ready_to_produce)
+        init_production_count = len(productions)
+        for production in productions:
+            production.action_check_with_batch()
```

Comment:
action_check_with_batch() should be a multi then do this here
```suggestion
        productions.action_check_with_batch()
```